### ISSUE TITLE
Transfer learning using dataset with different keypoints

### ISSUE THREAD

**Issue by axelssons** [TaatiTeam/MotionAGFormer#29]
Thank you for the great work with MotionAGFormer!

I have an own dataset with a different set of keypoints compared to H36M, with additional keypoints on the toes. Note that this dataset is small compared to H36M. The goal is to use the pre-trained (on H36M) MotionAGFormer-B model and train/finetune it on my own dataset using transfer learning, getting it to output my custom set of keypoints. How would one go about freezing the convolutional base layers and replacing the final fully connected layer / head layer with new ones having random weights on which we train? I assume this is a good option to solve the problem.

Best regards,
Jonathan

**Comment by SoroushMehraban** [comment]
Hi @axelssons,
I don't think it's easy to do it. Specially in MotionAGFormer one of the streams (GCNFormer) uses BatchNorm and changing the number of frames/keypoints might be problematic. I think one possible solution to your problem is to pretrain it on a dataset that also covers toe keypoints such as [H3WB](https://github.com/wholebody3d/wholebody3d). For doing so:
- Note that this dataset also covers face/hand keypoints. You can discard them if you don't need them.
- Also note that the keypoints are in a different format. So you have to change the `CONNECTIONS` [here](https://github.com/TaatiTeam/MotionAGFormer/blob/master/model/modules/graph.py#L6C1-L6C12) to create a new adjacency matrix

After pretraining on this dataset, you can fine-tune on yours and hopefully you'll get promising results 🙂 note that in our code we also used `2.5d_factor`  that I explained [here](https://github.com/TaatiTeam/MotionAGFormer/issues/4#issuecomment-1847441992). In a nutshell, it's impossible to extract keypoint scales having a single camera. Because person tall far from camera and person short close to camera both are having the same 2D sequence scales. Therefore we use this `2.5d_factor` to get rid of the scale issue and model is more focused to estimate joint kinematics instead. So you have to define it in a preprocessing step based on what I explained in that issue.
